src/code/code_800F07C0.c

```diff
@@ -1,31 +1,261 @@
+        enHy->actor.objBankIndex = enHy->unk192;
+        isInitialized = true;
+        ActorShape_Init(&enHy->actor.shape, 0.0f, NULL, 0.0f);
+        gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[enHy->actor.objBankIndex].segment);
+        SkelAnime_InitFlex(globalCtx, &enHy->skelAnime, skeletonHeaderSeg, NULL, enHy->jointTable, enHy->morphTable,
+                           16);
+        EnHy_ChangeObjectAndAnim(enHy, globalCtx, animIndex);
+    }
+    return isInitialized;
+}
+
+//! @TODO: Should just take EnDoor instead of actor when c and h are split
+void func_800F0BB4(EnHy* enHy, GlobalContext* globalCtx, Actor* door, s16 arg3, s16 arg4) {
+    s32 pad;
+    s8 sp3B;
+    Vec3f offset;
+    f32 phi_f0;
+
+    Actor_CalcOffsetOrientedToDrawRotation(door, &offset, &enHy->actor.world.pos);
+    phi_f0 = (offset.z >= 0.0f) ? 1.0f : -1.0f;
+    sp3B = ((s8)phi_f0 < 0) ? 0 : 2;
+    EnHy_ChangeObjectAndAnim(enHy, globalCtx, (sp3B == 0) ? arg3 : arg4);
+    enHy->skelAnime.baseTransl = *enHy->skelAnime.jointTable;
+    enHy->skelAnime.prevTransl = *enHy->skelAnime.jointTable;
+    enHy->skelAnime.moveFlags |= 3;
+    AnimationContext_SetMoveActor(globalCtx, &enHy->actor, &enHy->skelAnime, 1.0f);
+    ((EnDoor*)door)->unk_1A1 = 1;
+    ((EnDoor*)door)->unk_1A0 = sp3B;
+}
+
+s32 func_800F0CE4(EnHy* enHy, GlobalContext* globalCtx, ActorFunc draw, s16 arg3, s16 arg4, f32 arg5) {
+    s32 ret = false;
+    s16 yaw;
+    Actor* door;
+    s32 pad;
+
+    if (func_8013D68C(enHy->path, enHy->curPoint, &enHy->actor.world.pos)) {
+        door = EnHy_FindNearestDoor(&enHy->actor, globalCtx);
+        if (door != NULL) {
+            ret = true;
+            func_800F0BB4(enHy, globalCtx, door, arg3, arg4);
+            yaw = Math_Vec3f_Yaw(&enHy->actor.world.pos, &door->world.pos);
+            enHy->actor.world.pos.x += arg5 * Math_SinS(yaw);
+            enHy->actor.world.pos.z += arg5 * Math_CosS(yaw);
+            enHy->actor.world.rot.y = -yaw;
+            enHy->actor.shape.rot.y = -yaw;
+            enHy->actor.draw = draw;
+        }
+    }
+    return ret;
+}
+
+s32 func_800F0DD4(EnHy* enHy, GlobalContext* globalCtx, s16 arg2, s16 arg3) {
+    s32 ret = false;
+    s32 pad;
+    Actor* door;
+
+    enHy->curPoint = 0;
+    if (func_8013D68C(enHy->path, enHy->curPoint, &enHy->actor.world.pos)) {
+        door = EnHy_FindNearestDoor(&enHy->actor, globalCtx);
+        if (door != NULL) {
+            ret = true;
+            func_800F0BB4(enHy, globalCtx, door, arg2, arg3);
+            enHy->actor.shape.rot.y = Math_Vec3f_Yaw(&enHy->actor.world.pos, &door->world.pos);
+            enHy->actor.world.rot.y = enHy->actor.shape.rot.y;
+            enHy->actor.gravity = 0.0f;
+            enHy->actor.flags &= ~1;
+        }
+    }
+    return ret;
+}
+
+s32 func_800F0E94(EnHy* enHy, GlobalContext* globalCtx, f32 gravity, s16 animIndex) {
```

Comment:
```suggestion
s32 EnHy_SetNextPoint(EnHy* enHy, GlobalContext* globalCtx, f32 gravity, s16 animIndex) {
```
maybe?

Reply:
I do like the suggestion. Though I think it would be best to keep it in theme with the move fucntions below. Perhaps something like `EnHy_SetPointFowards` and `EnHy_SetPointBackwards`?

Reply:
Sounds good.